.github/workflows/ubuntu-n.yaml

```diff
@@ -7,20 +7,286 @@ on:
+          if ( ! $? ) {
+            exit 1
+          }
+
+          Write-Output "Starting Ubuntu 24.04 and waiting for cloud-init to finish..."
+          Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList "--distribution $DistroName --user root --cd /root --exec cloud-init status --wait"
+          if ( ! $? ) {
+            exit 1
+          }
+
+          Write-Output "Validating cloud-init schema."
+          Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList "--distribution $DistroName --user root --cd /root --exec cloud-init schema --system"
+          if ( ! $? ) {
+            exit 1
+          }
+
+      - name: Cache snaps necessary for testing on amd64
+        id: cache-snaps
+        uses: actions/cache@v4
+        with:
+          path: snaps/
+          key: snaps-amd64
+
+      - name: Download snaps necessary for testing on amd64
+        if: steps.cache-snaps.outputs.cache-hit != 'true'
+        run: |
+          $DistroName = 'Test-Ubuntu-24.04'
+          Write-Output "Downloading snaps necessary for testing..."
+          if ( -not ( Test-Path -PathType Container -LiteralPath "snaps" ) ) {
+            New-Item -ItemType Directory -Name "snaps"
+            if ( ! $? ) {
+              exit 1
+            }
+          }
+
+          Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList "--distribution $DistroName --user root --cd ~ --exec snap download --target-directory=/srv/workspace/snaps snapd"
+          if ( ! $? ) {
+            exit 1
+          }
+
+          Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList "--distribution $DistroName --user root --cd ~ --exec snap download --target-directory=/srv/workspace/snaps bare"
+          if ( ! $? ) {
+            exit 1
+          }
+
+          Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList "--distribution $DistroName --user root --cd ~ --exec snap download --target-directory=/srv/workspace/snaps core22"
+          if ( ! $? ) {
+            exit 1
+          }
+
+          Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList "--distribution $DistroName --user root --cd ~ --exec snap download --target-directory=/srv/workspace/snaps hello"
+          if ( ! $? ) {
+            exit 1
+          }
+
+          Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList "--distribution $DistroName --user root --cd ~ --exec snap download --target-directory=/srv/workspace/snaps bare"
+          if ( ! $? ) {
+            exit 1
+          }
+
+          Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList "--distribution $DistroName --user root --cd ~ --exec snap download --target-directory=/srv/workspace/snaps lxd"
+          if ( ! $? ) {
+            exit 1
+          }
```

Comment:
Same about error messages here

Reply:
I've added error messages to all the commands now.